### PR TITLE
add test workflow

### PR DESCRIPTION
**Pull request by sensuikan1973**
run unit test on PR/push.
this improve PR quality check, I think.

example: https://github.com/sensuikan1973/cloud_firestore_mocks/pull/1

regards,

### CODE REVIEW

File: .github/workflows/widget_test.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      - master
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
```
Comment (atn832) [inline .github/workflows/widget_test.yaml:17]: Looks like fetch-depth defaults to 1. Can you remove this line?

https://github.com/actions/checkout

File: .github/workflows/widget_test.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+name: Unit_Test
```
Comment (atn832) [inline .github/workflows/widget_test.yaml:1]: Is it possible to use spaces in the name? Like "Unit tests". This way it'll look human readable when we add a status badge.

See https://help.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow#adding-a-workflow-status-badge-to-your-repository

Comment (atn832) [reply .github/workflows/widget_test.yaml:1]: It looks like it's indeed possible: https://help.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow#workflow-file-example

**Comment by atn832** [comment]
Can you also rename the file to `unit-tests.yaml`? To use the same dash separated filenames as the GitHub documentation.

**Comment by sensuikan1973** [comment]
fixed! 🚀

**Comment by atn832** [comment]
Thank you @sensuikan1973, this is great!

Fixes #19.

**Pull request merged**